AsyncDisplayKit/ASDisplayNode.mm

```diff
@@ -703,6 +704,89 @@ - (instancetype)styledWithBlock:(void (^)(ASLayoutElementStyle *style))styleBloc
 
 #pragma mark - Layout
 
+- (void)invalidateSize
+{
+  ASDisplayNodeAssertThreadAffinity(self);
```

Comment:
It's probably best to keep this thread-affined for now.  However, it's exciting to note that this only reads self.bounds, so we could potentially make it threadsafe in the future (.threadsafeBounds)